### PR TITLE
Remove ability to add configuration as a dependency - it is now an error

### LINKED ISSUE #21724: Remove @Deprecated DefaultDependencyHandler.doAdd(Conf, Obj, Closure)

### PR DESCRIPTION
**Pull request by tresat**
Fixes #21724

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Review by jvandort — CHANGES_REQUESTED** [review]

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java [hunk 1]
```diff
@@ -174,14 +175,8 @@ public Dependency create(Object dependencyNotation, @Nullable Closure configureC
...
         if (dependencyNotation instanceof Configuration) {
-            DeprecationLogger.deprecateBehaviour("Adding a Configuration as a dependency is a confusing behavior which isn't recommended.")
-                .withAdvice("If you're interested in inheriting the dependencies from the Configuration you are adding, you should use Configuration#extendsFrom instead.")
-                .willBeRemovedInGradle8()
-                .withDslReference(Configuration.class, "extendsFrom(org.gradle.api.artifacts.Configuration[])")
-                .nagUser();
-            return doAddConfiguration(configuration, (Configuration) dependencyNotation);
-        }
-        if (dependencyNotation instanceof ProviderConvertible<?>) {
+            throw new GradleException("Adding a Configuration as a dependency is a confusing behavior that is no longer allowed as of Gradle 8.0.");
```
[the remark this answers is not in the document]
Comment (tresat) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java:178]: I was keeping the existing wording, but on reflection I agree with you, "behavior" isn't great here.  I think I'll just remove that "confusing behavior" bit completely and just say it is no longer allowed.

**Comment by tresat** [comment]
@bot-gradle test this

**Review by jvandort — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**